### DISCUSSION TITLE
Invalid Bundle Format; Signature Size exeeds bundle size error.

### DISCUSSION THREAD

**Discussion by umarFarooq-emb** [rauc/rauc#1388] [General]
Hello,

I am working on protecting the rootfs with dm-verity. After I integerated this in the system my rootfs bundle stopped working. I have set the rootfs slot as raw in the system.conf and also my rootfs name is `*.ext4.sign-verity`. Hash table is also included in the image being packaged in the bundle.

Initially is was getting the following errors.

```
root@-v4-00000000:/tmp/rauc# rauc install core-image-minimal.raucb
installing
  0% Installing
  0% Determining slot states
 20% Determining slot states done.
 20% Checking bundle
 20% Verifying signature
 40% Verifying signature done.
[54570.418878] loop0: detected capacity change from 0 to 342171
 40% Checking bundle done.
[54570.457351] loop0: detected capacity change from 342171 to 342168
[54570.511739] device-mapper: verity: sha256 using implementation "stm32-sha256"
 40% Checking manifest contents
 60% Checking manifest contents done.  
 60% Determining target install group  
 80% Determining target install group done.
[54570.697229] device-mapper: ioctl: remove_all left 1 open device(s)
 80% Updating slots
 99% Updating slots failed.
100% Installing failed.
LastError: Installation error: Unsupported image /run/rauc/bundle/icore-image-minimal.ext4
.sign-verity for slot type raw
Installing `/tmp/rauc/core-image-minimal.raucb` failed
```

I created the following patch and applied.

```diff
diff --git a/src/update_handler.c b/src/update_handler.c
index 37587bd..9085776 100644
--- a/src/update_handler.c
+++ b/src/update_handler.c
@@ -2500,6 +2500,7 @@ RaucUpdatePair updatepairs[] = {
        {"*.ext4.caibx", "ext4", img_to_fs_handler},
        {"*.ext4.caibx", "raw", img_to_raw_handler},
        {"*.vfat.caibx", "raw", img_to_raw_handler},
+       {"*.ext4.sign-verity", "raw", img_to_raw_handler},
        {"*.ubifs.caibx", "ubivol", img_to_ubivol_handler},
        {"*.ubifs.caibx", "ubifs", img_to_ubifs_handler},
        //{"*.img.caibx", "nand", img_to_nand_handler}, /* unsupported */
```

Now the following error is appearing...

```
root@v4-00000000:/tmp/umar# rauc install core-image-minimal.raucb
installing
  0% Installing
  0% Determining slot states
 20% Determining slot states done.
 20% Checking bundle
 40% Checking bundle failed.
100% Installing failed.
LastError: Invalid bundle format: Signature size (9110166045834546007) exceeds bundle size
Installing `/tmp/umar/core-image-minimal.raucb` failed
```

I would really appreciate any assistance on this.

Thanks

Regards,
Umar

**Comment by ejoerns** [comment]
> I created the following patch and applied.

If you can rename the image file before creating the bundle, this would be the better option.

> LastError: Invalid bundle format: Signature size (9110166045834546007) exceeds bundle size

This really looks like a broken bundle. How did you create it? Does it look valid on the host (when using `rauc info`)?

**Reply by umarFarooq-emb** [reply]
If I rename the file before creating the bundle I get the following logs `signature is 0`

and Rauc info also says the same thing,, signature size exceeds the bundle size.

And I have studied the manifest,, the it looks fine.

> I have not added ext4.tar.sign-verity;; I suppose tar is not supported with raw slots.

**Reply by ejoerns** [reply]
Renaming input images should not affect signature creation. Do you move the *bundle* before finalizing creation or something like that? Does the creation process succeed without errors?

**Reply by umarFarooq-emb** [reply]
I just clean the bundle using `cleanall` in bitbake process and the bundle creation is completed without errors.

**Reply by umarFarooq-emb** [reply]
```
root@-v4-00000000:/# rauc install /tmp/umar/core-image-minimal.raucb 
installing
  0% Installing
  0% Determining slot states
 20% Determining slot states done.
 20% Checking bundle
 40% Checking bundle failed.
100% Installing failed.
LastError: Invalid bundle format: Signature size is 0
idle
Installing `/tmp/umar/core-image-minimal.raucb` failed
```

I have created the bundle again and I am getting the above logs

**Reply by umarFarooq-emb** [reply]
```
[update]
compatible=minimal-machine
version=1.0
description=minimal-bundle version 1.0-r0
build=20240416095134

[bundle]
format=verity

[image.rootfs-minmal-os]
sha256=998a891068d5569dde9af48857079fd341317a3ddd6bb13482ace3f1dd4d8ab6
size=646828032
filename=core-image-minimal.ext4.sign-verity
```
this is the manifest file

**Comment by umarFarooq-emb** [comment]
```
RAUC_SLOT_rootfs-minimal-os[fstype] = "ext4.sign-verity"
```

Adding this line in the bundle.bb I can update the rootfs successfully. Thanks for your help.

**Reply by ejoerns** [reply]
Note that this should **not** have something to do with the 'signature size' errors.

Maybe the bundle creation failed silently somehow? Would be worth a look into the log.do_bundle file.
